utils/buttons.py

```diff
@@ -3,58 +3,92 @@
 import asyncio
 import contextlib
 import inspect
-import os
 import time
 
 from copy import copy
 from enum import Enum
 from functools import partial
-from typing import (TYPE_CHECKING, Any, AsyncGenerator, Callable, Coroutine, Dict, Iterable, List, Optional, Tuple, Type,
-                    Union)
+from typing import (TYPE_CHECKING, Any, AsyncGenerator, Awaitable, Callable, Dict, Iterable, List, Optional, Type,
+                    TypeVar, Union)
 
+import asyncpg
 import discord
 
 from discord import ui
 from discord.ext import commands
-from discord.ui import TextInput
 
 from utils.context_managers import UserLock
 from utils.menus import ListPageInteractionBase, MenuBase, MenuViewInteractionBase
 from utils.useful import StellaEmbed
 
 if TYPE_CHECKING:
+    from main import StellaBot
     from utils.useful import StellaContext
 
 
+T = TypeVar("T")
+R = TypeVar("R")
+
+InteractionCallback = Callable[[discord.Interaction], Awaitable[None]]
+
+
 class BaseButton(ui.Button):
-    def __init__(self, *, style: discord.ButtonStyle, selected: Union[int, str], row: int,
-                 label: Optional[str] = None, **kwargs: Any):
+    def __init__(self, *, style: discord.ButtonStyle, selected: Union[int, str] = "", row: Optional[int] = None,
+                 label: Optional[str] = None, stay_active: bool = False, **kwargs: Any):
         super().__init__(style=style, label=label or selected, row=row, **kwargs)
         self.selected = selected
+        self.stay_active = stay_active
```

Comment:
Property name is still up to debate. I had the following ideas: `always_active, always_on, stay_on, permanent`